Terminal.Gui/Views/TextView.cs

```diff
@@ -1,19 +1,82 @@
+#nullable enable
+
 // TextView.cs: multi-line text editing
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Globalization;
 using System.IO;
 using System.Linq;
 using System.Runtime.CompilerServices;
 using System.Text;
+using System.Text.Json.Serialization;
 using System.Threading;
 using Terminal.Gui.Resources;
 
 namespace Terminal.Gui {
+	/// <summary>
+	/// Represents a single row/column within the <see cref="TextView"/>. Includes the glyph and the foreground/background colors.
+	/// </summary>
+	[DebuggerDisplay ("{DebuggerDisplay}")]
+	public class RuneCell : IEquatable<RuneCell> {
+		/// <summary>
+		/// The glyph to draw.
+		/// </summary>
+		[JsonConverter (typeof (RuneJsonConverter))]
+		public Rune Rune { get; set; }
+
+		/// <summary>
+		/// The foreground color to draw the glyph with.
+		/// </summary>
+		[JsonConverter (typeof (ColorSchemeJsonConverter))]
+		public ColorScheme? ColorScheme { get; set; }
+
+		/// <summary>Indicates whether the current object is equal to another object of the same type.</summary>
+		/// <param name="other">An object to compare with this object.</param>
+		/// <returns>
+		///   <see langword="true" /> if the current object is equal to the <paramref name="other" /> parameter; 
+		///   otherwise, <see langword="false" />.</returns>
+		public bool Equals (RuneCell? other)
+		{
+			return other != null &&
+			    Rune.Equals (other.Rune) &&
+			    ColorScheme == other.ColorScheme;
+		}
+
+		/// <summary>Returns a string that represents the current object.</summary>
+		/// <returns>A string that represents the current object.</returns>
+		public override string ToString ()
+		{
+			string colorSchemeStr = ColorSchemeDebuggerDisplay ();
+			return DebuggerDisplay;
+		}
+
+		private string ColorSchemeDebuggerDisplay ()
```

Comment:
@tig I tried all with `public overrides string ToString ()` but the Rune was never showing as I like when debugging. For e.g. '\n' was showing as only '' and this can be like a space, a tab or other control ascii control. The only solution I found for this was using the `DebuggerDisplay`, see here (https://learn.microsoft.com/en-us/visualstudio/debugger/using-the-debuggerdisplay-attribute?view=vs-2022).
This is a private method and I decided to mark it as it's using the `DebuggerDisplay`.

Reply:
Ah. I see. I didn't know about [DebuggerDisplay]!